src/main/java/com/commercetools/sync/categories/CategorySync.java

```diff
@@ -0,0 +1,199 @@
+import com.commercetools.sync.categories.utils.CategorySyncUtils;
+import com.commercetools.sync.commons.Sync;
+import com.commercetools.sync.services.CategoryService;
+import com.commercetools.sync.services.TypeService;
+import com.commercetools.sync.services.impl.CategoryServiceImpl;
+import com.commercetools.sync.services.impl.TypeServiceImpl;
+import io.sphere.sdk.categories.Category;
+import io.sphere.sdk.categories.CategoryDraft;
+import io.sphere.sdk.commands.UpdateAction;
+import io.sphere.sdk.models.SphereException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.List;
+
+import static java.lang.String.format;
+import static org.apache.commons.lang3.StringUtils.isNotBlank;
+
+public class CategorySync implements Sync<CategoryDraft, Category> {
+    private static final Logger LOGGER = LoggerFactory.getLogger(CategorySync.class);
+    private final CategorySyncOptions syncOptions;
+    private final CategorySyncStatistics statistics;
+
+    private final TypeService typeService;
+    private final CategoryService categoryService;
+
+    public CategorySync(@Nonnull final CategorySyncOptions syncOptions) {
+        this.syncOptions = syncOptions;
+        this.statistics = new CategorySyncStatistics();
+        this.typeService = new TypeServiceImpl(syncOptions.getCtpClient().getClient());
+        this.categoryService = new CategoryServiceImpl(syncOptions.getCtpClient().getClient());
+    }
+
+    CategorySync(@Nonnull final CategorySyncOptions syncOptions,
+                 @Nonnull final TypeService typeService,
+                 @Nonnull final CategoryService categoryService) {
+        this.syncOptions = syncOptions;
+        this.statistics = new CategorySyncStatistics();
+        this.typeService = typeService;
+        this.categoryService = categoryService;
+    }
+
+    /**
+     * Traverses a {@link List} of {@link CategoryDraft} objects and tries to fetch a category, from the CTP project with
+     * the configuration stored in the {@code syncOptions} instance of this class, using the external id. If a category exists,
+     * this category is synced to be the same as the new category draft in this list. If no category exist with such external id,
+     * a new category, identical to this new category draft  is created.
+     * <p>
+     * The {@code statistics} instance is updated accordingly to whether the CTP request was carried out successfully or not.
+     * If an exception was thrown on executing the request to CTP,
+     * the optional error callback specified in the {@code syncOptions} is called.
+     *
+     * @param categoryDrafts the list of new category drafts to sync to the CTP project.
+     */
+    @Override
+    public void syncDrafts(@Nonnull final List<CategoryDraft> categoryDrafts) {
+        LOGGER.info(format("About to sync %d category drafts into CTP project with key '%s'."
```

Comment:
This is the only log message from this library that would notify the user that it actually started.